### PR TITLE
JBPM-9204 - Make jbpm-work-items repository compile with JDK 11

### PR DESCRIPTION
**Pull request by MarianMacik**
I have managed to make this repo compilable with JDK 11 (mvn clean install -DskipTests). Tests work in all modules but two:

* mavenembedder-workitem because there are some issues with Eclipse JDK compiler
* karaf-itests because these tests use pax-exam-junit4 dependency which seems to be not working with JDK 11 in the version we use now

### CODE REVIEW

**Comment by MarianMacik** [comment]
jenkins execute full downstream build

**Comment by tsurdilo** [comment]
@MarianMacik I need to test start.jbpm.org app with these changes because it heavily uses the mavenembedder workitem. Please allow me some time to do that. Thanks.

File: camel-workitem/src/main/java/org/jbpm/process/workitem/camel/CXFCamelWorkitemHandler.java [hunk 1]
```diff
@@ -48,7 +49,8 @@
         },
         serviceInfo = @WidService(category = "${name}", description = "${description}",
                 keywords = "apache,camel,cxf,payload,route,connector",
-                action = @WidAction(title = "Connect to a JAX-WS service hosted in CXF")
+                action = @WidAction(title = "Connect to a JAX-WS service hosted in CXF"),
+                authinfo = @WidAuth
```
Comment (tsurdilo) [inline camel-workitem/src/main/java/org/jbpm/process/workitem/camel/CXFCamelWorkitemHandler.java:53]: Why is this needed ? Looks weird

Comment (MarianMacik) [reply camel-workitem/src/main/java/org/jbpm/process/workitem/camel/CXFCamelWorkitemHandler.java:53]: @tsurdilo Please see the discussion in GChat.

Comment (tsurdilo) [reply camel-workitem/src/main/java/org/jbpm/process/workitem/camel/CXFCamelWorkitemHandler.java:53]: I have lots of messages in GChat, also other chats. Could you maybe give a summary here. I can also do all my PR reviews on GChat as well if this is the way we are going?

Comment (MarianMacik) [reply camel-workitem/src/main/java/org/jbpm/process/workitem/camel/CXFCamelWorkitemHandler.java:53]: @tsurdilo Sorry, but the original intention was to point you to the whole thread we already have there about this topic...

Here is the copied summary:

When compiling with JDK 11 to JDK 8 target it fails with the message:

`Error post-processing workitem annotations: Incorrectly typed data found for annotation element public abstract org.jbpm.process.workitem.core.util.service.WidAuth org.jbpm.process.workitem.core.util.service.WidService.authinfo() (Found data of type org.jbpm.process.workitem.core.util.service.WidAuth)`

The issue is WidAuth String[] parameters (array of strings) - when I remove them it works 🙂 Also if I compile to JDK 11 target it works. In fact, it is even enough to add an import of WidAuth into the annotated file (WIH in this case) and it will start working. But in that way the import is redundant (and could be easily removed by mistake), so I decided to explicitly define @WidAuth which will solve this issue and there will be no effect on the code. I already have a reproducer which I will use in the issue I want to file in maven-compiler-plugin and if it gets forwarded to Java compiler, I will create an issue in Java compiler. Hope that helped.

Comment (tsurdilo) [reply camel-workitem/src/main/java/org/jbpm/process/workitem/camel/CXFCamelWorkitemHandler.java:53]: Thanks!! I guess this is fine then if otherwise does not compile.

Comment (MarianMacik) [reply camel-workitem/src/main/java/org/jbpm/process/workitem/camel/CXFCamelWorkitemHandler.java:53]: Yep, the issue is really weird, but it works now :)

**Comment by MarianMacik** [comment]
jenkins execute full downstream build

**Comment by MarianMacik** [comment]
jenkins execute full downstream build

**Comment by MarianMacik** [comment]
FDB changes not related. Can you approve @mareknovotny?

**Comment by mareknovotny** [comment]
fyi @ruivieira can you check the tests why there is 
`Unable to load datasource properties [/datasource.properties]` in both https://github.com/kiegroup/kie-jpmml-integration/blob/master/jbpm-recommendation-pmml-logistic-regression/src/test/java/org/jbpm/prediction/pmml/PMMLLogisticRegressionPredictionServiceProcessTest.java and https://github.com/kiegroup/kie-jpmml-integration/blob/master/jbpm-recommendation-pmml-random-forest/src/test/java/org/jbpm/prediction/pmml/PMMLRandomForestPredictionServiceProcessTest.java

**Comment by ruivieira** [comment]
@mareknovotny Checking it. Just a FYI, this is not something I've seen before, and there are no `datasource.properties` uses in that example.

**Comment by ruivieira** [comment]
@mareknovotny, @yesamer seems to be working already on a fix for this regression on https://github.com/kiegroup/kie-jpmml-integration/pull/32. This will fix KOGITO-2367 also (https://github.com/kiegroup/kie-wb-common/pull/3345)

**Comment by MarianMacik** [comment]
Fixes #134.
